src/frontend/python/chiquito/cb.py

```diff
@@ -222,3 +227,93 @@ def to_constraint(v: ToConstraint) -> Constraint:
         raise TypeError(
             f"Type `{type(v)}` is not ToConstraint (one of Constraint, Expr, int, or F)."
         )
+
+
+@dataclass
+class LookupTableRegistry:
+    tables: Dict[int, LookupTable] = field(default_factory=dict)
+
+    def add(self: LookupTableRegistry, table: LookupTable):
+        self.tables[table.uuid] = table
+
+    def get(self: LookupTableRegistry, uuid: int) -> LookupTable:
+        return self.tables[uuid]
+
+
+@dataclass
+class LookupTable:
```

Comment:
@qwang98 when a table is created, it should be close to add more expresions to dest. So perhaps there is a method create() that returns self and sets a flag that prevents from adding more things to dest. Also, the table cannot be used if this flag has not been activated by call to create()

Reply:
I believe this `create()` function you mentioned is actually `new_table()` in `dsl.py`. Rust Chiquito has the same naming.

I modified `LookupTable` such that now we have a new `read_only` flag, that defaults to `False` when `LookupTable` is just created. User can then call `add()` to add more destination columns to the `LookupTable`, which requires that `read_only == False`. 

`LookupTable` will then need to be added by `new_table()` in `dsl.py` to the dictionary of UUID to `LookupTable`. `new_table()` also changes the `read_only` flag to `True`, so that no more calls of `add()` to modify the destination column is possible after `new_table()` is called.

Finally, any use of the table will be through `apply()` and `when()`, both of which requires that `read_only == True`.

This should accomplish all the features you suggested above.

What's more? `create_table()` can only be called during `setup()` and there's a flag assertion for that as well.